src/utils/ContainerUtils.ts

```diff
@@ -0,0 +1,89 @@
+import { ContainerProps, ListView } from "../components/DropdownFilterContainer";
+
+export const parseStyle = (style = ""): {[key: string]: string} => {
+    try {
+        return style.split(";").reduce<{[key: string]: string}>((styleObject, line) => {
+            const pair = line.split(":");
+            if (pair.length === 2) {
+                const name = pair[0].trim().replace(/(-.)/g, match => match[1].toUpperCase());
+                styleObject[name] = pair[1].trim();
+            }
+            return styleObject;
+        }, {});
+    } catch (error) {
+        // tslint:disable-next-line no-console
+        window.console.log("Failed to parse style", style, error);
+    }
+
+    return {};
+};
+export class Utils {
+    static validate(props: ContainerProps & { filterNode: HTMLElement; targetListView: ListView; validate: boolean }): string {
+        const widgetName = "dropdown-filter";
+        // validate filter values if filterby = attribute, then value should not be empty or "" or " ".
+        if (!props.filterNode) {
+            return `${widgetName}: unable to find a listview with to attach to`;
+        }
+
+        if (!(props.targetListView && props.targetListView._datasource)) {
+            return `${widgetName}: this Mendix version is incompatible`;
```

Comment:
Please Check all private API structure.
Missing at least: `_datasource._constraints` and `targetListView.update`

What about `filter` and `entity`